### PR TITLE
Add an example integration test for SSE

### PR DESCRIPTION
**Pull request by chenhunghan**
## Motivation

The SSE example was missing a test, would be nice to have one.

## Solution

This PR adds an example for integration test.

- Adding the integration test requiring some minor refactoring of the example, this PR moves the create of app into a separate `app()`.
-  Add dev-dependencies: `reqwest`/`reqwest-eventsource`/`eventsource-stream` that were used in tests.

### CODE REVIEW

File: examples/sse/src/main.rs [hunk 1]
```diff
@@ -44,16 +43,35 @@ async fn main() {
...
+}
+
+#[derive(Deserialize)]
+struct Repeat {
+    take_n: Option<usize>,
+}
+
 async fn sse_handler(
     TypedHeader(user_agent): TypedHeader<headers::UserAgent>,
+    repeat: Query<Repeat>,
```
Comment (davidpdrsn) [inline examples/sse/src/main.rs:63]: Adding this specifically for testing feels wrong. I think the test should be able to just take a certain number of events itself. Perhaps using https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.take

Comment (chenhunghan) [reply examples/sse/src/main.rs:63]: It's using `.take` at line 74, maybe the question is how do you recommend to pass n to `.take()` in the test?

Comment (chenhunghan) [reply examples/sse/src/main.rs]: Like this? https://github.com/tokio-rs/axum/pull/2465/commits/9cc61f1318fc65b880b2bcc682cab98e2e6eb838

File: examples/sse/src/main.rs [hunk 2]
```diff
@@ -63,3 +70,59 @@ async fn sse_handler(
...
+    use eventsource_stream::Eventsource;
+    use tokio::net::TcpListener;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn integration_test() {
+        // A helper function that spawns our application in the background
+        async fn spawn_app(host: impl Into<String>) -> String {
+            let _host = host.into();
```
Comment (davidpdrsn) [inline examples/sse/src/main.rs:85]: Why the `_`?

```suggestion
            let host = host.into();
```

File: examples/sse/src/main.rs [hunk 3]
```diff
@@ -63,3 +70,59 @@ async fn sse_handler(
...
+            // Bind to localhost at the port 0, which will let the OS assign an available port to us
+            let listener = TcpListener::bind(format!("{}:0", _host)).await.unwrap();
+            // Retrieve the port assigned to us by the OS
+            let port = listener.local_addr().unwrap().port();
+            tokio::spawn(async {
+                let app = app();
+                axum::serve(listener, app).await.unwrap();
```
Comment (davidpdrsn) [inline examples/sse/src/main.rs:92]:
```suggestion
                axum::serve(listener, app()).await.unwrap();
```

**Review by davidpdrsn — COMMENTED** [review]
A new nits but otherwise looks good!

File: examples/sse/src/main.rs (hunk 1 above)
Comment (davidpdrsn) [reply examples/sse/src/main.rs]: Yes!

File: examples/sse/src/main.rs (hunk 3 above)
Comment (chenhunghan) [reply examples/sse/src/main.rs:92]: Good catch

**Review by davidpdrsn — APPROVED** [review]
Thanks!

**Pull request merged**